obs-studio-client/source/controller.cpp

```diff
@@ -117,12 +152,28 @@ bool close_process(ProcessInfo pi) {
 }
 
 std::string get_process_name(ProcessInfo pi) {
-	std::vector<char> name_buf(MAX_PATH + 2);
-	DWORD name_len = MAX_PATH + 1;
-	if (QueryFullProcessImageName((HANDLE)pi.handle, 0, name_buf.data(), &name_len)) {
-		return std::string(name_buf.data(), name_buf.data() + name_len);
+	/* 255 is a coincidental value actually given by GetVolumeInformation.
+	 * This value can be something other than 255 but 255 is the common
+	 * value. We're targetting desktops and the likelihood this turns into
+	 * an issue is almost none but should be noted at least. */
```

Comment:
At least for me, the maximum path allowed on C:\ is 32767, as returned by GetVolumeInformation()->lpMaximumComponentLength. Perhaps we should query GetVolumeInformation() here?

Reply:
I think you're confusing max path length with component length. Maximum path length on NTFS I believe is 255. Windows will generally not allow you to make a folder or file past somewhere along 236 characters though. See (this)[http://rextester.com/KJC3864] for an example output of GetVolumeInformation. 

If you want to make sure that's correct, that's fine, especially if someone installs onto a non-NTFS partitions (such as fat though I believe 255 there is still more than good enough).

Reply:
I am talking about the total path length, which can be bigger than 255 on NTFS (starting with Windows 7, NTFS2). This is a policy that can be enabled for all drives but by default is only enabled for C:\ (as far as i know at least). For example, this is a valid path in PowerShell and other Long Paths capable programs:
```
\\?\D:\aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
That's 742 characters for the entire path, far above the 255 character limit that started to stop existing with NTFS2.

Since `get_process_name` is not returning just the executable file name but actually the full binary path (for safety, probably should have renamed the function), this will break on something like this for example:

```
Works:
C:\Users\MichaelFabianDirks\AppData\Local\Programs\Streamlabs\Streamlabs OBS\app.asar.unpacked\node_modules\obs-studio-node\obs64.exe

Broken:
D:\Program Files (x86)\Automatisch Installiert\Meine Software\Freie Software\GPLv3\Media, Video & Audio\Streaming, Recording & Tools\Streamlabs\Streamlabs Open Broadcaster Software\Streamlabs Open Broadcaster Software (1)\app.asar.unpacked\node_modules\obs-studio-node\obs64.exe
```

Even though this case can be very unlikely, I think we should still handle that case and try with a 32kb buffer the 2nd time. Though the only apps I've seen actually fail with this (so far) are git, mercury/hg and chrome/firefox if the username is too long and the cache path exceeds 255 characters because of that.

Reply:
Seems we can do this for this functionality: http://rextester.com/KJC3864
I have a working version in code as well...